File: InterviewCakeProblems/inFlightEntertainment.js
function inFlightEntertainment(flight_length, movie_lengths) {
   let seenMovies = new Set();
   
   for (let i = 0; i < movie_lengths.length; i++) {
       let firstMovie = movie_lengths[i];
       let secondMovie = flight_length - firstMovie;
       if (seenMovies.has(secondMovie)) {
           return true;
       } else {
           seenMovies.add(firstMovie)
       }
   }
   return false;
}

console.log(inFlightEntertainment(100, [25, 30, 60, 75, 90, 80, 40, 70]))
console.log(inFlightEntertainment(100, [25, 30, 60, 85, 90, 80, 50]))

File: python-practice/inflight_entertainment.py
def inflight_entertainment(flight_length, movie_lengths):
    seen = set()
    for first_movie_len in movie_lengths:
        # first_movie_len = movie_lengths[m]
        second_movie_len = flight_length - first_movie_len
        if second_movie_len in seen:
            return True
        first_movie_len.add(first_movie_len)
    return False
